src/views/MainView.vue

```diff
@@ -27,74 +16,56 @@ import PollViewer from '../components/PollViewer/PollViewer.vue'
 import TopBar from '../components/TopBar/TopBar.vue'
 
 import { useIsInCall } from '../composables/useIsInCall.js'
-import Router from '../router/router.js'
-
-export default {
-	name: 'MainView',
-	components: {
-		CallView,
-		CallFailedDialog,
-		ChatView,
-		LobbyScreen,
-		PollViewer,
-		TopBar,
-	},
-
-	props: {
-		token: {
-			type: String,
-			required: true,
-		},
-	},
-
-	setup() {
-		const isInCall = useIsInCall()
-		return { isInCall }
-	},
+import { useStore } from '../composables/useStore.js'
 
-	computed: {
-		conversation() {
-			return this.$store.getters.conversation(this.token)
-		},
+const props = defineProps<{
+	token: string
+}>()
 
-		isInLobby() {
-			return this.$store.getters.isInLobby
-		},
+const store = useStore()
+const isInCall = useIsInCall()
+const router = useRouter()
+const route = useRoute()
 
-		connectionFailed() {
-			return this.$store.getters.connectionFailed(this.token)
-		},
-	},
+const isInLobby = computed(() => store.getters.isInLobby)
+const connectionFailed = computed(() => store.getters.connectionFailed(props.token))
 
-	watch: {
-		isInLobby(isInLobby) {
-			// User is now blocked by the lobby
-			if (isInLobby && this.isInCall) {
-				this.$store.dispatch('leaveCall', {
-					token: this.token,
-					participantIdentifier: this.$store.getters.getParticipantIdentifier(),
-				})
-			}
-		},
-	},
+watch(isInLobby, (isInLobby) => {
+	// User is now blocked by the lobby
+	if (isInLobby && isInCall.value) {
+		store.dispatch('leaveCall', {
+			token: props.token,
+			participantIdentifier: store.getters.getParticipantIdentifier(),
+		})
+	}
+})
 
-	mounted() {
-		const handleRouteHashChange = (token, route) => {
-			if (route?.hash === '#direct-call') {
-				emit('talk:media-settings:show')
-				Router.replace({ ...route, hash: '' })
-			} else if (route?.hash === '#settings') {
-				emit('show-conversation-settings', { token })
-				Router.replace({ ...route, hash: '' })
-			}
+onMounted(() => {
+	watchEffect(() => {
+		if (route.hash === '#direct-call') {
+			emit('talk:media-settings:show', '')
+			router.replace({ hash: '' })
+		} else if (route.hash === '#settings') {
+			emit('show-conversation-settings', { token: store.state.token })
```

Comment:
I'm afraid to forget about it during some store migration:

```suggestion
			emit('show-conversation-settings', { token: store.getters.getToken() })
```